pkg/api/types.go

```diff
@@ -742,12 +742,25 @@ type Service struct {
 }
 
 // Endpoints is a collection of endpoints that implement the actual service, for example:
-// Name: "mysql", Endpoints: ["10.10.1.1:1909", "10.10.2.2:8834"]
+// Name: "mysql", Endpoints: [{"ip": "10.10.1.1", "port": 1909}, {"ip": "10.10.2.2", "port": 8834}]
 type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
-	Endpoints []string `json:"endpoints,omitempty"`
+	// Optional: The IP protocol for these endpoints. Supports "TCP" and
+	// "UDP".  Defaults to "TCP".
+	Protocol  Protocol   `json:"protocol,omitempty"`
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
```

Comment:
The real fix (IMO) would be to rename the struct to ServiceEndpoints or something, but ain't nobody got time to rename REST objects :)

HostPorts is wrong because it implies something like Container[x].Ports[y].HostPort, which is something else.  PodPorts is wrong because endpoints are supposed to be only loosely coupled to kubernetes, and may not even be pods.  Targets?  Destinations?  I didn't find any of these to be significantly better than Endpoints, so I left it as it was.